Core/GDCore/Project/EventsFunction.cpp

```diff
@@ -59,11 +59,19 @@ const std::vector<gd::ParameterMetadata>& EventsFunction::GetParametersForEvents
 void EventsFunction::SerializeTo(SerializerElement& element) const {
   element.SetAttribute("name", name);
   element.SetAttribute("fullName", fullName);
-  element.SetAttribute("description", description);
+  if (!description.empty()) {
+    element.SetAttribute("description", description);
+  }
   element.SetAttribute("sentence", sentence);
-  element.SetAttribute("group", group);
-  element.SetAttribute("getterName", getterName);
-  element.SetBoolAttribute("private", isPrivate);
+  if (!group.empty()) {
+    element.SetAttribute("group", group);
+  }
+  if (!getterName.empty()) {
+    element.SetAttribute("getterName", getterName);
+  }
+  if (isPrivate) {
```

Comment:
is there a risk the setBoolAttribute is not triggered if we're serializing an object with isPrivate = false?

Reply:
Yes, this is the point.
`GetBoolAttribute` return `false` by default and `isPrivate` is `false` most of the time.

Reply:
Nice!